pkg/server/token_minter.go

```diff
@@ -109,13 +112,69 @@ func (s *TokenMintServer) handleToken() http.Handler {
 		}
 		auditEvent.HTTPStatusCode = respCode
 
-		// TODO: Write the audit information to Cloud Loggin
+		// Write the audit information
+		if err := s.writeAuditLog(ctx, &auditEvent); err != nil {
+			logger.ErrorContext(ctx, "failed to write audit event", "error", err)
+			// Fail the request if it could not be audited
+			respCode = http.StatusInternalServerError
+			respMsg = "failed to persist audit information"
+		}
 
 		w.WriteHeader(respCode)
 		fmt.Fprint(w, respMsg)
 	})
 }
 
+// writeAuditLog takes all of the data about the request and its success or failure and
+// writes it to cloud logging.
+func (s *TokenMintServer) writeAuditLog(ctx context.Context, auditEvent *auditEvent) error {
+	logger := logging.FromContext(ctx)
+
+	token, err := json.Marshal(auditEvent.Token)
+	if err != nil {
+		return fmt.Errorf("error marshaling token: %w", err)
+	}
+	req, err := json.Marshal(auditEvent.Request)
+	if err != nil {
+		return fmt.Errorf("error marshaling request: %w", err)
+	}
+	config, err := json.Marshal(auditEvent.Config)
+	if err != nil {
+		return fmt.Errorf("error marshaling config: %w", err)
+	}
+
+	resourceName := "not specified"
+	if auditEvent.Request != nil {
+		resourceName = strings.Join(auditEvent.Request.Repositories, ",")
+	}
+	principal := "unknown"
+	if auditEvent.Token != nil {
+		principal = auditEvent.Token.WorkflowRef
+	}
+
+	logRequest := &audit.AuditLog{
+		ServiceName:  "abcxyz.dev/github-token-minter",
+		MethodName:   "abcxyz.dev/github-token-minter/HandleTokenRequest",
+		ResourceName: resourceName,
+		Status: &status.Status{
+			Code:    int32(auditEvent.HTTPStatusCode),
+			Message: auditEvent.HTTPErrorMessage,
+		},
+		Request: &structpb.Struct{Fields: map[string]*structpb.Value{
+			"token":   structpb.NewStringValue(string(token)),
+			"request": structpb.NewStringValue(string(req)),
+			"config":  structpb.NewStringValue(string(config)),
+		}},
+		AuthenticationInfo: &audit.AuthenticationInfo{
+			PrincipalEmail: principal,
+		},
+	}
+
```

Comment:
Is there a reason we need to use the Audit Logging proto? This is just a standard JSON log, so we can use our own format. I'd prefer to avoid bringing in grpc and all these deps.

I think we really just want something like:

```go
logger.DebugContext(ctx, "generating token", 
  "request.token": token, 
  "request.config": config,
  "resource": resourceName,
  "principal": principal)
```

Reply:
Ack